crates/optimism/payload/src/builder.rs

```diff
@@ -677,3 +757,67 @@ where
         Ok(None)
     }
 }
+
+/// Extracts commitment from access list entries, pointing to 0x420..022 and validates them
+/// against supervisor.
+///
+/// If commitment present pre-interop tx rejected.
+///
+/// returns (is_valid, is_recoverable)
+/// is_recoverable to set to false if we can't recover from error, in case supervisor is down,
+/// connection is broken or transaction is invalid deterministically.
+pub fn is_cross_tx_valid<Primitives: OpPayloadPrimitives>(
```

Comment:
```suggestion
pub fn is_valid_cross_tx<Primitives: OpPayloadPrimitives>(
```
better semantics because we aren't sure that the input tx is a cross chain tx until this function body

for the same reason, we want to return `Option<bool>`

rn op-reth isn't using the second index in the return tuple, so leave this out for now and add it back in a later pr which uses this logic